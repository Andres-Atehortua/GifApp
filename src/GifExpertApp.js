import React, { useState } from "react";
import AddCategory from "components/AddCategories";
import GifGrid from "components/GifGrid";

const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
