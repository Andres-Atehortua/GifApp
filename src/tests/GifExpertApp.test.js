import React from "react";
import { shallow } from "enzyme";
import GifExpertApp from "GifExpertApp";

describe("Pruebas al componente <GifExpertApp />", () => {
  test("should match snapshot", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should show a list of categories", () => {
    const categories = ["Goku", "Marvel"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
