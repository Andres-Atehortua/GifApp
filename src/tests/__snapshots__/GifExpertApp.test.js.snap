// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <GifExpertApp /> should match snapshot 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategories
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas al componente <GifExpertApp /> should show a list of categories 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategories
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Goku"
      key="Goku"
    />
    <GifGrid
      category="Marvel"
      key="Marvel"
    />
  </ol>
</Fragment>
`;
